File: cpp/barang.cpp
#include <bits/stdc++.h>
#include "produk.cpp"
using namespace std;

class barang : public produk // Class barang
{
private: // Atribut
    string merek;
    int stok;
    int tahun;

public:
    // Constructor
    barang() : produk() {}

    // Getter
    string getMerek() const { return this->merek; }
    int getStok() const { return this->stok; }
    int getTahun() const { return this->tahun; }

    // Setter
    void setMerek(string merek) { this->merek = merek; }
    void setStok(int stok) { this->stok = stok; }
    void setTahun(int tahun) { this->tahun = tahun; }

    // Destructor
    ~barang() {}
};

File: java/Barang.java
public class Barang extends Produk 
{
    private String merek;
    private int stok;
    private int tahun;

    // Constructor
    public Barang() 
    {
        super();
        this.merek = "";
        this.stok = 0;
        this.tahun = 0;
    }

    // Getter
    public String getMerek() { return this.merek; }
    public int getStok() { return this.stok; }
    public int getTahun() { return this.tahun; }

    // Setter
    public void setMerek(String merek) { this.merek = merek; }
    public void setStok(int stok) { this.stok = stok; }
    public void setTahun(int tahun) { this.tahun = tahun; }
}